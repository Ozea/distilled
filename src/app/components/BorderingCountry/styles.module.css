.border {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  min-width: 150px;
  font-style: italic;
}

.img {
  border: 1px solid #ebebeb;
  object-fit: cover;
}