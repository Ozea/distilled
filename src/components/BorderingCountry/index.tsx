import Image from 'next/image'
import Card from '../Card'

import styles from './styles.module.css'

export default function BorderingCountry({ name, flag, population }: IBorderingCountry) {
  return (
    <Card className={styles.border} key={name.official}>
      <Image src={flag.url} alt={flag.alt} width={75} height={50} className={styles.img} />
      <strong>{name.common}</strong>
      <span>
        <strong>Population:</strong> {population}
      </span>
    </Card>
  )
}
