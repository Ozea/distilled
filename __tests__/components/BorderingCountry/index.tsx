import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

import BorderingCountry from '@/components/BorderingCountry'

const country: IBorderingCountry = {
  name: {
    common: 'Ireland',
    official: 'Republic of Ireland',
  },
  population: '5.033.000',
  flag: {
    alt: 'Ireland',
    url: 'https://flagcdn.com/ie.svg',
  },
}

describe('BorderingCountry', () => {
  it('matches the snapshot', () => {
    const { container } = render(<BorderingCountry {...country} />)
    expect(container).toMatchSnapshot()
  })
  it('renders an image', () => {
    render(<BorderingCountry {...country} />)

    const img = screen.getByRole('img')

    expect(img).toBeInTheDocument()
    expect(img).toHaveAttribute('alt', country.name.common)
    expect(img).toHaveAttribute('src', country.flag.url)
  })
})
