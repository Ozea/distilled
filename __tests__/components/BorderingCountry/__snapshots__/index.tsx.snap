// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderingCountry matches the snapshot 1`] = `
<div>
  <div
    class="card border"
  >
    <img
      alt="Ireland"
      class="img"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="https://flagcdn.com/ie.svg"
      style="color: transparent;"
      width="75"
    />
    <strong>
      Ireland
    </strong>
    <span>
      <strong>
        Population:
      </strong>
       
      5.033.000
    </span>
  </div>
</div>
`;
